lxd/container_lxc.go

```diff
@@ -1374,11 +1374,19 @@ func (c *containerLXC) initLXC(config bool) error {
 			relativeDestPath := strings.TrimPrefix(destPath, "/")
 			sourceDevPath := filepath.Join(c.DevicesPath(), fmt.Sprintf("unix.%s.%s", strings.Replace(k, "/", "-", -1), strings.Replace(relativeDestPath, "/", "-", -1)))
 
+			// Do not fail to start when the device doesn't need to
+			// exist.
+			opts := "none bind,create=file"
+			if m["required"] != "" || !shared.IsTrue(m["required"]) {
```

Comment:
That check is wrong, setting required=true would cause it to be treated as optional.

You want:

    if m["required"] != "" && !shared.IsTrue(m["required"]) {

Reply:
corrected